docs/config.md

```diff
@@ -267,3 +267,18 @@ Disables the p2p-circuit relay transport.
 - `EnableRelayHop`
 Enables HOP relay for the node. If this is enabled, the node will act as
 an intermediate (Hop Relay) node in relay circuits for connected peers.
+
+### `ConnMgr`
+Connection manager configuration.
+
+- `Type`
+Sets the type of connection manager to use, options are: `"none"` and `"basic"`.
+
+- `LowWater`
+LowWater is the minimum number of connections to maintain.
+
+- `HighWater`
+HighWater is the number of connections that, when exceeded, will trigger a connection GC operation.
+- `GracePeriod`
+GracePeriod is a time duration that new connections are immune from being closed by the connection manager.
```

Comment:
Unless I'm mistaken, this can trivially be used to DOS a node.

Reply:
I wouldnt call it trivially, but yes, you can use this to exceed a nodes maximum connection count and make a connection closing sweep not reach the low water mark.

Reply:
Sorry, for some reason, I assumed we had a hard "don't open new connections past the high water mark" limit (I was a bit sleep deprived due to jet lag...). That would have effectively taken the node offline.

---

However, still worried that this will be a way to monopolize a peer by tying up their connection quota with connections in the grace period. Basically, you can ensure that no connection lasts more than 10 seconds. Maybe don't count connections in the grace period?

Reply:
Yeah, connections in the grace period are basically not counted, see here: https://github.com/libp2p/go-libp2p-connmgr/blob/master/connmgr.go#L61

Reply:
I see, so we literally just skip them. Nevermind, LGTM.